tests/test_sqs/test_sqs.py

```diff
@@ -796,6 +801,134 @@ def test_send_receive_message_timestamps():
     int.when.called_with(approximate_first_receive_timestamp).shouldnt.throw(ValueError)
 
 
+@mock_sqs
+def test_send_receive_message_with_attribute_name_all():
+    sqs = boto3.resource("sqs", region_name="us-east-1")
+    client = boto3.client("sqs", region_name="us-east-1")
+    client.create_queue(QueueName="test-queue")
+    queue = sqs.Queue("test-queue")
+
+    body_one = "this is a test message"
+    body_two = "this is another test message"
+
+    queue.send_message(MessageBody=body_one)
+    queue.send_message(MessageBody=body_two)
+
+    messages = client.receive_message(
+        QueueUrl=queue.url, AttributeNames=["All"], MaxNumberOfMessages=2
+    )["Messages"]
+
+    message1 = messages[0]
+    message2 = messages[1]
+
+    message1["Body"].should.equal(body_one)
+    message2["Body"].should.equal(body_two)
+
+    message1.shouldnt.have.key("MD5OfMessageAttributes")
+    message2.shouldnt.have.key("MD5OfMessageAttributes")
+
+    message1["Attributes"].should.have.key("ApproximateFirstReceiveTimestamp")
+    message1["Attributes"]["ApproximateReceiveCount"].should.equal("1")
+    message1["Attributes"].should.have.key("SenderId")
+    message1["Attributes"].should.have.key("SentTimestamp")
+    message2["Attributes"].should.have.key("ApproximateFirstReceiveTimestamp")
+    message2["Attributes"]["ApproximateReceiveCount"].should.equal("1")
+    message2["Attributes"].should.have.key("SenderId")
+    message2["Attributes"].should.have.key("SentTimestamp")
+
+
+@mock_sqs
+def test_send_receive_message_with_attribute_name_receive_time():
```

Comment:
Don't know how easy it would be to combine all these methods with a pytest.mark.parametrize?
There is a lot of duplication here

Reply:
good point, I will give it a try